src/main/java/de/tum/in/www1/artemis/service/scheduled/NotificationScheduleService.java

```diff
@@ -112,4 +115,85 @@ private void scheduleNotificationForExercise(Exercise exercise) {
             log.error("Failed to schedule notification for exercise " + exercise.getId(), exception);
         }
     }
+
+    /// EXERCISE_SUBMISSION_ASSESSED
+
+    /**
+     * updateScheduling method for the notificationType EXERCISE_SUBMISSION_ASSESSED
+     * @param exercise that should trigger a notification when the assessment due date is over
+     */
+    public void updateSchedulingForAssessedExercisesSubmissions(Exercise exercise) {
+        checkSecurityUtils();
+        if (exercise.getAssessmentDueDate() == null || ZonedDateTime.now().isAfter(exercise.getAssessmentDueDate())) {
+            // to make sure no wrong notification is sent out the date is checked again in the concrete notification method
+            scheduleService.cancelScheduledTaskForLifecycle(exercise.getId(), ExerciseLifecycle.ASSESSMENT_DUE);
+            return;
+        }
+        if (exercise.isCourseExercise()) {
+            scheduleNotificationForAssessedExercisesSubmissions(exercise);
+        }
+    }
+
+    /**
+     * The place where the actual tasks/methods are called/scheduled that should be run when at the assessment due date of the exercise
+     * @param exercise which will be announced by a notifications at release date
+     */
+    private void scheduleNotificationForAssessedExercisesSubmissions(Exercise exercise) {
+        try {
+            checkSecurityUtils();
+            scheduleService.scheduleTask(exercise, ExerciseLifecycle.ASSESSMENT_DUE, () -> {
+                checkSecurityUtils();
+                Exercise foundCurrentVersionOfScheduledExercise = findCurrentVersionOfScheduledExercise(exercise);
+                if (foundCurrentVersionOfScheduledExercise == null) {
```

Comment:
see below, remove error handling, foundCurrentVersionOfScheduledExercise should never be null

Reply:
I also removed this error handling (in both methods)